hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/RSRpcServices.java

```diff
@@ -3167,7 +3187,18 @@ private RegionScannerHolder newRegionScanner(ScanRequest request, ScanResponse.B
     builder.setMvccReadPoint(scanner.getMvccReadPoint());
     builder.setTtl(scannerLeaseTimeoutPeriod);
     String scannerName = String.valueOf(scannerId);
-    return addScanner(scannerName, scanner, shipper, region, scan.isNeedCursorResult());
+
+    boolean isFullRegionScan = false;
+    if(!region.getRegionInfo().getTable().isSystemTable()) {
+      if(checkFullRegionScan(scan, region)){
+        isFullRegionScan = true;
+      } else {
+        isFullRegionScan = false;
+      }
+    } else {
+      isFullRegionScan = false;
```

Comment:
This might not be needed. anyway we have inited to false na.

Reply:
thx for the catch, yes fixed.